### ISSUE TITLE
Line breaks and tabs not working

### ISSUE THREAD

**Issue by dsinni** [lojjic/aframe-troika-text#9]
Hello!

Very promising component. I've just started experimenting with it and I noticed that line breaks and tabs are not supported with `\n` and `\t` within the `value` property.

```html
<a-entity troika-text="value: Hello\nWorld!;"></a-entity>
```

![image](https://user-images.githubusercontent.com/4733018/74089580-9d0eb200-4a70-11ea-8db3-44da10a8f2b5.png)

I noticed it was in the documentation, so was hoping it would function as expected.

Should this be currently supported?

Thanks!

**Comment by lojjic** [comment]
Oh, that's very interesting. The documentation is mostly copied from the underlying JS component, which does indeed support `\n` and `\t` because that is how you represent newlines and tabs in JavaScript. But since AFrame markup isn't JavaScript I suppose that wouldn't work.

Does it work if you use literal newlines and tabs in your markup? Or with the appropriate HTML entities for those characters?

**Comment by lojjic** [comment]
To answer my own question: Yes, literal newline and tab characters within the markup do appear to work (* see below). Using HTML entities does not -- it seems to break the parsing -- but I think that's related to how AFrame parses attribute values.

> (*) There appears to be another issue that makes newlines wrap on top of each other; this looks like a problem with how default `lineHeight` is applied, and I'll push a fix for that asap.

**Comment by lojjic** [comment]
I've pushed an update to the docs that clarifies using literal newline/tab characters in the HTML markup. I've also fixed the lineHeight default issue. Please let me know if you continue having issues with it.

And thank you for testing and providing feedback! I'm a fan of AFrame but don't use it in my day job so I haven't had much opportunity to dogfood this component. I appreciate help identifying ways to make it stronger. 😄

**Comment by dsinni** [comment]
Thanks for the feedback, @lojjic ! And sorry for the delayed response.

If it's at all helpful, A-Frame provides an API for parsing component properties through the schema: https://aframe.io/docs/1.0.0/core/component.html#custom-property-type

Being that the A-Frame `text` component handles the `\n` and `\t` characters as expected, that could possibly be a useful reference for implementing within your component.

If it's not something you think you'll get around to, I may tinker a bit with it in the future. I really think using Troika for text has quite a few advantages over A-Frame's core `text` component, however now all of them are yet implemented in this component. I think the `clipRect` would be a game-changer, but that's obviously unrelated to this issue.

Anyway, thanks!

**Comment by lojjic** [comment]
> Being that the A-Frame text component handles the \n and \t characters as expected

Oh! I didn't realize that. Well if the core text component does it, then there's no reason I shouldn't be able to. Thanks for the link, I'll take a look at that and how the core text component handles it and try to match.

(Update: well I looked at the source and they just do a regex replace in the component. Easy enough! 😆)

Thanks for the note about `clipRect`, I opened #10 for that. Not sure why I didn't expose it before.
